tests/integration/targets/aci_l3out/tasks/main.yml

```diff
@@ -0,0 +1,257 @@
+      <<: *aci_l3out_present
+      l3protocol: bgp
+    register: nm_add_l3out_bgp
+
+  - name: Add BGP protocol again - testing L3protocol changes
+    aci_l3out:
+      <<: *aci_l3out_present
+      l3protocol: bgp
+    register: nm_add_l3out_bgp_again
+
+  - name: Add PIM protocol, ospf_spec and asn - testing warning message for asn
+    aci_l3out:
+      <<: *aci_l3out_present
+      l3protocol: pim
+      asn: 1
+    register: nm_add_l3out_pim
+
+  - name: Add EIGRP protocol - testing warning message for OSPF spec
+    aci_l3out:
+      <<: *aci_l3out_present
+      l3protocol: eigrp
+      asn: 1
+      ospf:
+        description: OSPF warnings message works
+    register: nm_add_l3out_eigrp
+
+  - name: remove asn - testing failure message
+    aci_l3out:
+      <<: *aci_l3out_present
+      l3protocol: eigrp
+      asn: null
+    register: add_l3out_without_asn
+    ignore_errors: true
+  
+  - name: remove protocol - testing static protocol
+    aci_l3out:
+      <<: *aci_l3out_present
+      l3protocol: static
+    register: nm_remove_l3protocol
+
+  - name: asserts for l3out protocols tasks
+    assert:
+      that:
+        - nm_add_l3out_bgp is changed
+        - nm_add_l3out_bgp_again is changed
+        - nm_add_l3out_pim is changed
+        - nm_add_l3out_pim.warnings.0 == "Parameter 'asn' is only applicable when l3protocol is 'eigrp'. The ASN will be ignored"
+        - nm_add_l3out_eigrp is changed
+        - nm_add_l3out_eigrp.sent.l3extOut.children.1.eigrpExtP.attributes.asn == "1"
+        - nm_add_l3out_eigrp.warnings.0 == "Parameter 'ospf' is only applicable when l3protocol is 'ospf'. The OPSF specifications will be ignored"
+        - add_l3out_without_asn.msg == "Parameter 'asn' is required when l3protocol is 'eigrp'"
+        - nm_remove_l3protocol is changed
+
+  - name: Add OSPF protocol
+    aci_l3out:
+      <<: *aci_l3out_present
+      l3protocol: ospf
+      ospf:
+        area_cost: 1
+        area_ctrl: [ summary, redistribute ]
+        area_id: 0.0.0.1
+        area_type: regular
+        multipod_internal: no
+        description: test for ospf protocol
+    register: nm_add_l3out_ospf
+
+  - name: asserts for l3out OSPF protocol tasks
+    assert:
+      that:
+        - nm_add_l3out_ospf is changed
+        - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.areaCost == "1"
+        - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.areaCtrl == "summary,redistribute"
+        - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.areaId == "0.0.0.1"
+        - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.areaType == "regular"
+        - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.descr == "test for ospf protocol"
+        - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.multipodInternal == "no"
+
+  - name: get ansible_l3out
+    aci_l3out:
+      <<: *aci_l3out_present
+      state: query
+    register: get_l3out
```

Comment:
Can you add another assert to this one. If I understand the l3out should now have all external profiles children types configured. Could you assert on the children returned by query that this is actually the case?